mod/youtubevideo/lib.php

```diff
@@ -32,8 +33,32 @@ function youtubevideo_delete_instance($id)
     if (!$youtubevideo = $DB->get_record('youtubevideo', array('id' => $id))) {
         return false;
     }
+    $DB->delete_records('youtubevideo', array('id' => $youtubevideo->id));
+    return true;
+}
 
-    $DB->delete_records('youtubevideo', array('id' => $id));
+function youtubevideo_supports($feature)
+{
+    switch($feature) {
+        case FEATURE_MOD_INTRO: return true;
+        case FEATURE_SHOW_DESCRIPTION: return true;
+        case FEATURE_GRADE_HAS_GRADE: return false;
+        case FEATURE_BACKUP_MOODLE2: return true;
+        default: return null;
+    }
+}
 
-    return true;
+function get_youtube_id($url)

```

Comment:
Une proposition plus élégante.
```suggestion
function get_youtube_id($url)
{
    $parts = parse_url($url);

    // Vérifie la présence de paramètres dans l'URL
    if (!empty($parts['query'])) {
        parse_str($parts['query'], $qs);
        return $qs['v'] ?? $qs['vi'] ?? false;  // Récupère l'ID 'v' ou 'vi' ou retourne false
    }

    // Vérifie la présence du chemin dans l'URL
    if (!empty($parts['path'])) {
        $path = explode('/', trim($parts['path'], '/'));
        return end($path);  // Retourne le dernier segment du chemin
    }

    return false;
}
```